packages/nightly-connect/src/lib/nightly-connect.ts

```diff
@@ -0,0 +1,223 @@
+import { AppNear, Network, NightlyConnectModal } from "@nightlylabs/connect";
+import { NearAppInfo } from "@nightlylabs/connect/lib/sdk/src/types/AppInfo";
+import type {
+  WalletModuleFactory,
+  WalletBehaviourFactory,
+  BridgeWallet,
+  Optional,
+  Transaction,
+} from "@near-wallet-selector/core";
+import { signTransactions } from "@near-wallet-selector/wallet-utils";
+import { PublicKey } from "near-api-js/lib/utils";
+import { Signer, utils, transactions as nearTransactions } from "near-api-js";
+
+const setupNightlyConnectState = async (
+  params: NearAppInfo
+): Promise<AppNear> => {
+  const client = await AppNear.build(params);
+  return client;
+};
+
+export interface NightlyConnectParams {
+  application: string;
+  description: string;
+  appIcon: string;
+  additionalInfo?: string;
+  url?: string;
+  timeout?: number;
+}
+const NightlyConnect: WalletBehaviourFactory<
+  BridgeWallet,
+  { params: NightlyConnectParams }
+> = async ({ store, params, logger, options, provider }) => {
+  let connectedAccounts: Array<{ accountId: string; publicKey: PublicKey }> =
+    [];
+  let client: AppNear | undefined;
+  const modal = new NightlyConnectModal();
+
+  const getAccounts = () => {
+    return connectedAccounts;
+  };
+  const signOut = async (): Promise<void> =>
+    new Promise<void>((resolve, reject) => {
+      const timer = setTimeout(() => {
+        reject("Closing connection timed out");
+      }, 5000);
+      client!.ws.onclose = () => {
+        clearTimeout(timer);
+        client = undefined;
+        connectedAccounts = [];
+        resolve();
+      };
+      client?.ws.close();
+    });
+
+  const transformTransactions = (
+    transactions: Array<Optional<Transaction, "signerId" | "receiverId">>
+  ): Array<Transaction> => {
+    const accounts = getAccounts();
+    const { contract } = store.getState();
+
+    if (!accounts.length || !contract) {
+      throw new Error("Wallet not signed in");
+    }
+
+    return transactions.map((transaction) => {
+      return {
+        signerId: transaction.signerId || accounts[0].accountId,
```

Comment:
We'll need to update this to use the new active account feature instead of defaulting to the first account: https://github.com/near/wallet-selector/pull/328

Reply:
fixed